Monitoring of docker images - admission controller (#2771)
kyma-project/test-infra: development/image-guard/README.md

> # image-guard

Reviewer: ```suggestion
# Image-guard
```

Reviewer: Actually, this is the title of the doc so I'd leave it capitalized 😉

This directory contains source code for the Image-guard admission controller.

> After installing the chart Image-guard performs the following tasks:

Reviewer: ```suggestion
Image-guard performs the following tasks:
```

> * Monitor and log used Docker images in the Pods.

Reviewer: ```suggestion
* Monitor and log used Docker images in the Pods.
```

> * Monitor and log used docker images in the pods

Reviewer: ```suggestion
* Monitor and log used Docker images in the Pods.
```

> * Optionally check and allow usage only for the whitelisted registries.

Reviewer: ```suggestion
* Optionally, check and allow the usage of the whitelisted registries only.
```

> * Optionally check and allow usage only for whitelisted registries

Reviewer: ```suggestion
* Optionally check and allow usage only for the whitelisted registries.
```

> ## Requirements

Reviewer: ```suggestion
## Prerequisites 
```

> To deploy the application you need the following:

Reviewer: ```suggestion
To deploy the application, you need the following:
```

> * helm 3

Reviewer: ```suggestion
* Helm 3
```

> * K8s cluster version at least 1.16

Reviewer: ```suggestion
* Kubernetes cluster (at least 1.16 version)
```

## Installation

> To install image-guard use the following command using your Kubernetes cluster context:

Reviewer: ```suggestion
To install image-guard, use the following command using your Kubernetes cluster context:
```

> To install image-guard use the following command using your k8s cluster context:

Reviewer: ```suggestion
To install image-guard, use the following command using your Kubernetes cluster context:
```

Reviewer: `Image-guard` or `image-guard` or maybe `Image Guard`? 🤔

Reviewer 2: I think it's best to stay with using `image-guard` as it's been loosely named to use it as name for the deployment.

```shell script
$ helm install image-guard ./image-guard
```

> If you want to enable enforcing image registries use helm flag `--set enforcedRegistry.enabled=true` and define the registries.

Reviewer: ```suggestion
If you want to enable enforcing image registries, use the `--set enforcedRegistry.enabled=true` Helm flag and define the registries.
```

## Flags
```
--host -h [host]                The hostname for the service

> --port -p [port]                The port to listen on (HTTPS).

Reviewer: ```suggestion
--port -p [port]                The HTTP port to listen on 
```

Reviewer 2: it needs to be a HTTPS port

> --port -p [port]                The HTTPS port to listen on.

Reviewer: ```suggestion
--port -p [port]                The HTTPS port to listen on
```

> --http-only                     Only listen on unencrypted HTTP (e.g. for proxied environments)

Reviewer: ```suggestion
--http-only                     The flag that enables listening on unencrypted HTTP only (for example: for proxied environments)
```

--key-path [path-to-file]       The path to the unencrypted TLS key
--cert-path [path-to-file]      The path to the PEM-encoded TLS certificate

> --allowed-registry -r [string]  Name of allowed registry

Reviewer: ```suggestion
--allowed-registry -r [string]  The name of the allowed registry
```

> --allowed-registry -r [string]  The name of the of allowed registry

Reviewer: ```suggestion
--allowed-registry -r [string]  The name of the allowed registry
```
